blockchain-source/src/cardano/event.rs

```diff
@@ -157,3 +145,51 @@ impl<Tip> GetNextFrom for CardanoNetworkEvent<BlockEvent, Tip> {
         }
     }
 }
+
+impl BlockEvent {
+    pub(crate) fn from_serialized_block(raw_block: SerializedBlock) -> anyhow::Result<Self> {
+        let block = raw_block
+            .unserialize()
+            .context("failed to deserialize block");
+
+        if let Ok(block) = block {
+            let id = BlockId::new(block.header().hash().to_string());
```

Comment:
little optimization: any time  `.header()` is called a new copy is made (header() does clone inside), we can make a variable for header and call the methods on it